source/dotnet/Test/AdaptiveCards.Test/SerializationTests.cs

```diff
@@ -496,6 +496,130 @@ public void BackgroundImage()
             Assert.AreEqual(expected, card.ToJson());
         }
 
+        [TestMethod]
+        public void RichTextBlock()
+        {
+            var card = new AdaptiveCard("1.2");
+
+            var richTB = new AdaptiveRichTextBlock();
+            richTB.Wrap = true;
+            richTB.MaxLines = 3;
+            richTB.HorizontalAlignment = AdaptiveHorizontalAlignment.Center;
+
+            // Build First Paragraph
+            var paragraph1 = new AdaptiveRichTextBlock.AdaptiveParagraph();
+
+            var textRun1 = new AdaptiveRichTextBlock.AdaptiveParagraph.AdaptiveTextRun("Start the first paragraph ");
+            paragraph1.Inlines.Add(textRun1);
+
+            var textRun2 = new AdaptiveRichTextBlock.AdaptiveParagraph.AdaptiveTextRun("with some cool looking stuff");
+            textRun2.Color = AdaptiveTextColor.Accent;
+            textRun2.FontStyle = AdaptiveFontStyle.Monospace;
+            textRun2.IsSubtle = true;
+            textRun2.Size = AdaptiveTextSize.Large;
+            textRun2.Weight = AdaptiveTextWeight.Bolder;
+            paragraph1.Inlines.Add(textRun2);
+
+            richTB.Paragraphs.Add(paragraph1);
+
+            // Build Second Paragraph (Empty inlines)
+            var paragraph2 = new AdaptiveRichTextBlock.AdaptiveParagraph();
+            richTB.Paragraphs.Add(paragraph2);
+
+            card.Body.Add(richTB);
+
+            var expected = @"{
```

Comment:
>var expected = @"{ [](start = 12, length = 18)

as noted in person, I think we would benefit from sample cards. be clear about when missing inlines or paragraphs become an error. also, show off markdown in inline runs! :)